### PR TITLE
Added some documentation on implementation and views

### PR DESCRIPTION
**Pull request by uaalto**
As per our discussion @oxtoacart @aranhoide

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -386,6 +453,36 @@ TODO - explain how subqueries work
...
+geo_country: AU   success_count: [2000, 2500]
+```
+
+One important aspect is that each field is actually not just an array, the array is actually preceded by the timestamp high water mark, so it would actually be something like this:
+
+`geo_country: US   success_count: 20180118T05:00Z[1000, 1500]`
+
+### Views
+
+Since everything in zenodb comes in through input streams, views cannot actually be constructed from the underlying tables, so they need to be stored independently. This allows for views to have different granularities that the tables/views they are referring to. In other words, at runtime, views are actually just like tables that pull from that same input stream, the only difference is that when you define a view, it can take into account knowledge from the definition of the underlying table.
```
Comment (aranhoide) [inline README.md]: Can views get data directly from the input stream, or only from other views/tables?

Comment (oxtoacart) [reply README.md]: What @uaalto is trying to explain here is that views actually can get data only from the input stream. It's impossible to retroactively populate a view using data from the underlying table. Views are really just a language construct for defining a table that's derived from another table.

File: README.md [hunk 2]
```diff
@@ -371,7 +371,74 @@ sort of stuff with data from thousands of servers and millions of clients!
 
 ZenoDB relies on a schema file (by default `schema.yaml`).
 
-TODO - fill this out
+### Example: How to add a view
+
+Defining a table as a view means that the data for this table comes from another table or view, rather than an input stream.
```
Comment (oxtoacart) [inline README.md]: @uaalto Yesterday you mentioned that a view is really just a language construct for creating a table whose properties are derived from an existing table. I think that's probably a better way of explaining this.

**Comment by oxtoacart** [comment]
Thanks for putting this together @uaalto! The only thing I would change is your description of what a view is.

**Comment by oxtoacart** [comment]
Thanks for doing this @uaalto!

**Pull request merged**